### PR TITLE
fix: Import grpc only for type checking in errors.py

### LINKED ISSUE #4344: Update CI to have a test for main feast dependency before release/deployment
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
As seen in https://github.com/feast-dev/feast/issues/4293, the PyPi package failed because of a dependency in a feast installation that uses an [extra](https://packaging.python.org/en/latest/tutorials/installing-packages/#installing-setuptools-extras).

```bash
pip install feast[sqlite_vec]
```

This can be added to the existing [build_wheels ci](https://github.com/feast-dev/feast/blob/master/.github/workflows/build_wheels.yml)  by testing a python import. If the import fails, we should fail the build/release.

**Describe alternatives you've considered**
We should have the CI test building the main feast package before release/deployment as well each extra/add-on.

**Additional context**

### LINKED ISSUE #4532: Running feast version command throw error
## Expected Behavior 
Should return the version without GRPCIO dependencies installed. 

## Current Behavior

Requires GRPCIO dependencies to be installed otherwise throws error 

`python3.11/site-packages/feast/errors.py", line 8, in <module>
    from grpc import StatusCode as GrpcStatusCode
ModuleNotFoundError: No module named 'grpc'`

## Steps to reproduce

Install latest feast and run `feast version` command

### Specifications

- Version: Master branch 
- Platform: Mac
- Subsystem:

## Possible Solution

1. Lazy imports with Try and Except
2. Either make GRPCIO dependencies as required (or)

### PR DESCRIPTION
**Pull request by tokoko**
Fixes #4532
Also fixes #4344

### CODE REVIEW

**Comment by dmartinol** [comment]
should we also mention the new make targets in the dev guide [document](https://github.com/feast-dev/feast/blob/7535b4036ce980c9a05bc33a9e61a7938ea1303e/docs/project/development-guide.md?plain=1#L157-L165)?

**Comment by tokoko** [comment]
@franciscojavierarceo Also added a test for `import feast` with no optional dependencies as part of this PR addressing #4344.

File: .github/workflows/smoke_tests.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: smoke-tests
```
Comment (franciscojavierarceo) [inline .github/workflows/smoke_tests.yml:1]: image

**Comment by tokoko** [comment]
@franciscojavierarceo this should be good now, no more proto generation headaches

**Review by franciscojavierarceo — APPROVED** [review]

**Pull request merged**